compiler/crates/relay-bin/src/main.rs

```
@@ -306,13 +314,69 @@ async fn handle_compiler_command(command: CompileCommand) -> Result<(), Error> {
     Ok(())
 }
 
+struct ExtraDataProvider {
+    script: String,
+}
+
+impl ExtraDataProvider {
+    pub fn new(script: String) -> ExtraDataProvider {
+        ExtraDataProvider {
+            script
+        }
+    }
+}
+
+impl LSPExtraDataProvider for ExtraDataProvider {
+    fn fetch_query_stats(&self, _search_token: &str) -> Vec<String> {
+        vec![]
+    }
+
+    fn resolve_field_definition(
+        &self,
+        project_name: String,
+        parent_type: String,
+        field_info: Option<FieldSchemaInfo>,
+    ) -> Result<Option<FieldDefinitionSourceInfo>, String> {
+        let entity_name = match field_info {
+            Some(field_info) => format!("{}.{}", parent_type, field_info.name),
+            None => parent_type,
+        };
+        let result = Command::new(&self.script)
+            .arg(project_name)
+            .arg(entity_name)
+            .output()
+            .expect("Failed to run extra data provider script");
+
+        let result = String::from_utf8(result.stdout).expect("Failed to parse output");
+
+        // Parse file_path:line_number:column_number
+        let result = result.trim();
+        let result = result.split(':').collect::<Vec<_>>();
+        if result.len() != 3 {
+            return Ok(None);
```

This function returns a result. Can we return an Err here?